.github/workflows/publish-algorithm-images.yaml

```diff
@@ -0,0 +1,99 @@
+name: Publish AutoML Algorithm Images
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Since this workflow runs when push event for master branch (when PR merged), We couldn't prevent docker build failed case previously.

How about make another template with `push:false` option, and run such events for `on.pull_requests`?
```
      - name: Build and Push
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ${{ inputs.dockerfile }}
          push: false <<< only build
```

This could make sure that master branch is always stable for building docker image.

Reply:
@anencore94 I think our AWS CI workflow verifies that all images can be built: https://github.com/kubeflow/katib/blob/master/test/e2e/v1beta1/argo_workflow.py#L327-L339.
Which means PRs that are merged to the `master` has valid docker build.
Does it make sense ?

Reply:
Yes, I didn't know there were an AWS CI workflows.
If so, how about create an **AWS CI-success badge** to master branch's README.md, if it is possible to integrate ?

Reply:
Hi, @anencore94.
We can see E2E running in AWD CI [here](https://argo.kubeflow-testing.com/workflows/).
For example, E2E for this PR is [this](https://argo.kubeflow-testing.com/workflows/kubeflow-test-infra/kubeflow-katib-presubmit-e2e-v1beta1-1746-bc60821-8000-497b).